src/main/java/com/vmtapp/enterprise/TicketController.java

```diff
@@ -95,10 +99,41 @@ public ResponseEntity fetchTicketsByAssignee(@PathVariable("assignee") String as
         return new ResponseEntity(HttpStatus.OK);
     }
 
-    @RequestMapping("/ticket/{id}")
-    public Optional<Ticket> fetchTicketById(@PathVariable("id") String id) throws Exception {
+    @GetMapping("/ticket/{id}")
+    public ModelAndView fetchTicketById(@PathVariable("id") int id) throws Exception {
+        ModelAndView modelAndView = new ModelAndView();
+
+
+        try{
+            // This checks if the ticket exists, and if not, throws an exception
+            Optional<Ticket> optionalTicket = Optional.ofNullable(ticketService.fetchTicketById(id).orElseThrow(
+                    () -> new Exception("not found")
+            ));
+
```

Comment:
One highlight: Optional allows you to check if object was found or not. In this case. if it wasn't found I am throwing an exception.